### PR TITLE
EVG-18525: Replace Antd Popconfirm

### PR DESCRIPTION
**Pull request by sophstad**
EVG-18525

### Description

- Replace Antd's popconfirm with our own implementation
- I added an uncontrolled version of Popconfirm that accepts just a trigger and internally manages the open state. This was easier when implementing components where the popconfirm element is mapped, which would have required tracking a lot of states and refs in the parent component.
- ...should we just use [Tooltip](https://www.mongodb.design/component/tooltip/example/) instead of Popconfirm? It seems to have a lot of features we would benefit from.

### CODE REVIEW

File: src/components/Popconfirm/index.tsx [hunk 1]
```diff
@@ -1,50 +1,78 @@
...
   confirmText = "Yes",
   refEl,
   onCancel = () => {},
   onConfirm = () => {},
   setActive,
+  trigger,
 }) => {
+  const [uncontrolledOpen, uncontrolledSetOpen] = useState<boolean>(false);
+  const setOpen =
+    (typeof active === "boolean" && setActive) || uncontrolledSetOpen;
```
Comment (khelif96) [inline src/components/Popconfirm/index.tsx:56]: Can you explain this type check? 
couldn't you do this?
```suggestion
    (active === null" && setActive) || uncontrolledSetOpen;
```

File: src/components/Popconfirm/index.tsx [hunk 2]
```diff
@@ -73,17 +102,45 @@ const Popconfirm: React.VFC<PopconfirmProps> = ({
...
+
+    return (
+      <>
+        {renderedTrigger}
+        {popoverContent}
+      </>
+    );
+  }
+
+  return popoverContent;
```
Comment (khelif96) [inline src/components/Popconfirm/index.tsx:129]:
```suggestion
  const triggerClickHandler = (e: React.MouseEvent) => {
    if (open) {
      onCancel();
    }

    setOpen(!open);
    trigger?.props?.onClick?.(e);
  };

  const renderedTrigger =
    trigger !== undefined &&
    cloneElement(trigger, {
      ref: triggerRef,
      onClick: triggerClickHandler,
    });

  return (
    <>
      {renderedTrigger}
      {popoverContent}
    </>
  );
```

**Review by khelif96 — CHANGES_REQUESTED** [review]
Great job taking this on!

While I do agree the tooltip serves our needs, making it interactable goes against leafy greens design guidelines. 

https://www.mongodb.design/component/tooltip/guidelines/#:~:text=Never%20nest%20call%20to%20actions%20within%20tooltips.
> Never nest call to actions within tooltips.

**Comment by sophstad** [comment]
> While I do agree the tooltip serves our needs, making it interactable goes against leafy greens design guidelines.

@khelif96 I noticed this too, but I feel like what we're doing by using the Popover here is effectively going against the guideline anyway. We're not _technically_ using Tooltip for an interaction, but we're effectively doing the same thing. Those guidelines also refer to mouseover instead of click interactions, so I wonder if they're a little outdated. Interested in any other thoughts on this!

**Comment by khelif96** [comment]
> > While I do agree the tooltip serves our needs, making it interactable goes against leafy greens design guidelines.
> 
> @khelif96 I noticed this too, but I feel like what we're doing by using the Popover here is kinda going against the guideline anyway. We're not _technically_ using Tooltip for an interaction, but we're effectively doing the same thing. I think it's a reasonable UX pattern regardless! Those guidelines also refer to mouseover instead of click interactions, so I wonder if they're a little outdated. Interested in any other thoughts on this!

That is a good point I would be curious to hear from the design folks if they have any opinions on it. But it does look like cloud does use popover to define similar interactions like this. 
https://github.com/search?q=repo:10gen/mms+@leafygreen-ui/popover&type=code
while the few tooltip usages I found did not define any interactivity. (Although I didn't look at all of them)
https://github.com/search?q=repo:10gen/mms+@leafygreen-ui/tooltip&type=code&p=2

**Comment by sophstad** [comment]
> That is a good point I would be curious to hear from the design folks if they have any opinions on it. But it does look like cloud does use popover to define similar interactions like this. https://github.com/search?q=repo:10gen/mms+@leafygreen-ui/popover&type=code while the few tooltip usages I found did not define any interactivity. (Although I didn't look at all of them) https://github.com/search?q=repo:10gen/mms+@leafygreen-ui/tooltip&type=code&p=2

Sounds good, I added to design sync for this week since this would make sense as a followup ticket anyway. Thanks!

File: src/components/Popconfirm/index.tsx [hunk 3]
```diff
@@ -1,50 +1,78 @@
...
   children: React.ReactNode;
+  confirmDisabled?: boolean;
   confirmText?: string;
-  refEl?: React.RefObject<HTMLElement>;
+  "data-cy"?: string;
+  justify?: Justify;
   onCancel?: () => void;
-  onConfirm?: () => void;
-  setActive: (open: boolean) => void;
+  onConfirm?: (e?: React.MouseEvent) => void;
```
[the remark this answers is not in the document]
Comment (sophstad) [reply src/components/Popconfirm/index.tsx:34]: `onConfirm` is called when the "ok" button is clicked, what kind of handler were you thinking of?

**Review by khelif96 — APPROVED** [review]
SGTM

**Comment by sophstad** [comment]
evergreen retry

**Pull request merged**